src/main/java/com/productdock/library/gateway/client/InventoryClient.java

```diff
@@ -9,25 +9,37 @@
 @Component
 public class InventoryClient {
 
-    @Value("${inventory.service.url}/api/inventory/book/")
+    @Value("${inventory.service.url}")
     private String inventoryServiceUrl;
 
     private WebClient webClient;
 
-    public InventoryClient(){
+    public InventoryClient() {
         this.webClient = WebClient.create();
     }
 
-    public Mono<Integer> getAvailableBookCopiesCount(String bookId, String jwtToken){
-        var inventoryBookUrl = inventoryServiceUrl + bookId;
+    public Mono<Integer> getAvailableBookCopiesCount(String bookId, String jwtToken) {
+        var inventoryBookUrl = inventoryServiceUrl + "/api/inventory/book/" + bookId;
 
         return webClient
                 .get()
                 .uri(inventoryBookUrl)
                 .header(HttpHeaders.AUTHORIZATION, "Bearer " + jwtToken)
                 .retrieve()
                 .bodyToMono(Integer.class)
-                .onErrorReturn(RuntimeException.class,0);
+                .onErrorReturn(RuntimeException.class, 0);
+    }
+
+    public Mono<Boolean> getBookSubscription(String bookId, String jwtToken) {
+        var subscriptionUrl = inventoryServiceUrl + "/api/inventory/subscriptions/" + bookId;
+
+        return webClient
+                .get()
+                .uri(subscriptionUrl)
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + jwtToken)
+                .retrieve()
+                .bodyToMono(Boolean.class)
+                .onErrorReturn(RuntimeException.class, false);
```

Comment:
If we create endpoint like this `/api/inventory/subscriptions/{bookId}`, by only reading it I would guess that it should return all book subscriptions, but I assume that you return if logged in user is subscribed or not